### PR TITLE
feat: add comprehensive unit tests for chain_data modules

### PR DESCRIPTION
**Pull request by 0xsatoshi99**
### Requirements for Adding, Changing, or Removing a Feature

* [x] Fill out the template below
* [x] The pull request must update the test suite to exercise the updated functionality
* [x] After you create the pull request, all status checks must pass

### Description of the Change

This PR adds comprehensive unit tests for multiple `bittensor.core.chain_data` modules that currently lack test coverage. The tests follow the existing testing patterns in the repository and use `unittest.mock.patch` for mocking dependencies.

**Modules tested:**
- `delegate_info.py` - DelegateInfoBase, DelegateInfo, DelegatedInfo classes
- `delegate_info_lite.py` - DelegateInfoLite class
- `neuron_info.py` - NeuronInfo with weights/bonds handling
- `neuron_info_lite.py` - NeuronInfoLite with metrics
- `stake_info.py` - StakeInfo with balance handling
- `subnet_info.py` - SubnetInfo with network parameters

**Test coverage includes:**
- Dataclass creation with valid data
- Edge cases (zero values, large values, empty collections)
- `_from_dict` deserialization methods
- Balance type handling and conversions
- Attribute access and validation
- Comparison between instances

### Alternate Designs

Considered using pytest fixtures but opted for helper functions to maintain consistency with existing test patterns in the repository.

### Possible Drawbacks

None identified. These are additive tests that do not modify existing functionality.

### Verification Process

1. All test files pass syntax validation:
```bash
python3 -m py_compile tests/unit_tests/chain_data/*_unit.py
```

2. Tests follow existing patterns from `tests/unit_tests/chain_data/test_metagraph_info.py`

3. Each test class covers:
   - Basic creation scenarios
   - Edge cases with boundary values
   - Deserialization from dict format
   - Type validation for Balance objects

### Release Notes

N/A - Internal test coverage improvement only.

### Branch Acknowledgement
[x] I am acknowledging that I am opening this branch against `staging`

---

*Contribution by Gittensor, learn more at https://gittensor.io/*

### CODE REVIEW

File: tests/unit_tests/chain_data/delegate_info_lite_unit.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+            validator_permits=[1, 2],
+            return_per_1000=Balance.from_tao(0.5),
+            total_daily_return=Balance.from_tao(100.0),
+        )
+        assert delegate.delegate_ss58 == MOCK_HOTKEY
+        assert delegate.take == 0.18
+        assert delegate.nominators == 100
+        assert delegate.owner_ss58 == MOCK_OWNER
+        assert delegate.registrations == [1, 2, 3]
+        assert delegate.validator_permits == [1, 2]
```
Comment (thewhaleking) [inline tests/unit_tests/chain_data/delegate_info_lite_unit.py:27]: what does this test?

File: tests/unit_tests/chain_data/delegate_info_lite_unit.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+            take=0.0,
+            nominators=0,
+            owner_ss58=MOCK_OWNER,
+            registrations=[],
+            validator_permits=[],
+            return_per_1000=Balance.from_rao(0),
+            total_daily_return=Balance.from_rao(0),
+        )
+        assert delegate.nominators == 0
+        assert delegate.take == 0.0
```
Comment (thewhaleking) [inline tests/unit_tests/chain_data/delegate_info_lite_unit.py:41]: What does this test?

File: tests/unit_tests/chain_data/delegate_info_lite_unit.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+            delegate_ss58=MOCK_HOTKEY,
+            take=0.18,
+            nominators=10000,
+            owner_ss58=MOCK_OWNER,
+            registrations=[1],
+            validator_permits=[1],
+            return_per_1000=Balance.from_tao(1.0),
+            total_daily_return=Balance.from_tao(10000.0),
+        )
+        assert delegate.nominators == 10000
```
Comment (thewhaleking) [inline tests/unit_tests/chain_data/delegate_info_lite_unit.py:54]: What does this test?

**Review by thewhaleking — COMMENTED** [review]
This is probably the fifteenth extremely low-quality PR I've reviewed this week.

**Comment by thewhaleking** [comment]
Another low-quality PR
